src/contracts/core/DelegationManager.sol

```diff
@@ -732,6 +748,32 @@ contract DelegationManager is
         _beaconChainSlashingFactor[staker] = bsf;
     }
 
+    /// @dev Calculate amount of slashable shares that are queued withdrawn from an operator for a strategy.
+    function _getSlashableSharesInQueue(
+        address operator,
+        IStrategy strategy,
+        uint64 maxMagnitude
+    ) internal view returns (uint256) {
+        uint256 currCumulativeScaledShares = uint256(_cumulativeScaledSharesHistory[operator][strategy].latest());
+        // Note: this will simply return 0 if no history exists, same for latest() as well
+        uint256 pastCumulativeScaledShares = _cumulativeScaledSharesHistory[operator][strategy].upperLookupRecent(
+            block.number - MIN_WITHDRAWAL_DELAY_BLOCKS
+        );
+        uint256 slashableShareInQueue = SlashingLib.scaleSharesForBurningQueueWithdrawals(
+            pastCumulativeScaledShares, currCumulativeScaledShares, maxMagnitude
+        );
+
+        return slashableShareInQueue;
+    }
+
+    /// @dev Update the cumulative withdrawn scaled shares from an operator for a given strategy
+    function _updateCumulativeScaledShares(address operator, IStrategy strategy, uint256 scaledShares) internal {
+        if (strategy != beaconChainETHStrategy) {
```

Comment:
Why would beacon chain stuff also not be added?

Do we add it to this array eventually when we have burning for native ETH?

Reply:
At minimum, in the `burnOperatorShares` method we wouldn't be able to use these cumulative scaled shares to efficiently burn beaconChain shares in a callback like the one we make to the StrategyManager.
However, the scaled shares does take into account slashing factors and therefore the bcsf. We could potentially store this value as well?